core/src/typecheck/unif.rs

```diff
@@ -1091,16 +1109,30 @@ impl PropagateConstrs for UnifRecordRows {
 }
 
 impl PropagateConstrs for UnifEnumRows {
-    fn propagate_constrs(&self, constr: &mut RowConstr, var_id: VarId) -> Result<(), RowUnifError> {
+    fn propagate_constrs(
+        &self,
+        constr: &mut RowConstrs,
+        var_id: VarId,
+    ) -> Result<(), RowUnifError> {
         fn propagate(
-            constr: &mut RowConstr,
+            constr: &mut RowConstrs,
             var_id: VarId,
             var_constr: HashSet<Ident>,
             erows: &UnifEnumRows,
         ) -> Result<(), RowUnifError> {
             match erows {
                 UnifEnumRows::Concrete {
-                    erows: EnumRowsF::Extend { row, .. },
+                    // If the row is an enum tag (ie `typ` is `None`), it can't cause any conflict.
+                    // See the documentation of RowConstrs for more details.
```

Comment:
Do we want to use a GHC style `[^row-tag-constraints]` for these? The same reference seems to appear a few times.

Reply:
I opted for rust-doc style `[RowConstrs]`, even if it's not checked in local comments. The rationale was that the documentation of `RowConstrs` is rust doc-style starting with `///`, and I didn't want to pollute it with a `[^row-tag-constraints]`. Assuming that the rustdoc form `[RowConstrs]` is easily understood and gotoref-able for any Rust developer.